cf/api/service_brokers.go

```diff
@@ -56,7 +56,7 @@ func (repo CloudControllerServiceBrokerRepository) FindByName(name string) (serv
 			return false
 		})
 
-	if !foundBroker {
+	if !foundBroker && (apiErr == nil) {
```

Comment:
Also, since you have now added another variable to the conditional, I expected at least two more tests or really four totals: `foundBroker and apiErr nil`, "foundBroker and apiErr not nil`, `!foundBroker and apiErr nil`, and `!foundBroker and apiErr not nil`.

Are you compacting the tests or am I missing something?